ios/ReactNativeAudioToolkit/ReactNativeAudioToolkit/AudioPlayer.m

```diff
@@ -203,12 +209,17 @@ - (void)invokeCallbackForPlayer:(ReactPlayer *)player {
     NSNumber *playerId = [self keyForPlayer:player];
     RCTResponseSenderBlock callback = self.callbackPool[playerId];
 
+    // If theres no callback that means we've already called the callback
+    // for this player, and have since disposed of it.
     if (!callback) {
         return;
     }
 
-    if (player.status == AVPlayerStatusReadyToPlay) {
+    CMTimeRange timeRange = player.currentItem.loadedTimeRanges.firstObject.CMTimeRangeValue;
+    Float64 loadedSeconds = CMTimeGetSeconds(timeRange.duration);
+    if (loadedSeconds > 10) {
```

Comment:
Its this kind of thing that could prevent us from contributing this fix upstream; 10 seconds is kinda arbitrarily picked and might not work for everyone. Could make it a setting that gets passed in like `autoDestroy`? If we want to contribute back upstream that is.

Reply:
Sounds like upstream isn't going to get much maintenance love, so maybe we're fine https://github.com/futurice/react-native-audio-toolkit/issues/56#issuecomment-334936599. Although I'd still quite like our fork to be a bit closer in line with theirs, for when things inevitably need to change with new versions of iOS and React Native.